subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeCompileTask.java

```diff
@@ -96,6 +98,18 @@ public String call() throws Exception {
                 return NativeToolChainInternal.Identifier.identify(toolChain, targetPlatform);
             }
         });
+        getOutputs().doNotCacheIf("No header dependency analysis provided", new Spec<Task>() {
+            @Override
+            public boolean isSatisfiedBy(Task element) {
+                return !AbstractNativeCompileTask.this.getHeaderDependenciesFile().isPresent();
+            }
+        });
+        getOutputs().doNotCacheIf("Native caching is not enabled", new Spec<Task>() {
```

Comment:
I think this should come first.

Reply:
I changed the order.

Reply:
Shall we say "Experimental native caching is not enabled"?